amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/logging/DiagnosticEventInput.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package software.amazon.kinesis.coordinator.logging;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.experimental.Accessors;
+import software.amazon.kinesis.leases.LeaseCoordinator;
+
+import java.util.concurrent.ThreadPoolExecutor;
+
+/**
+ * Container for the parameters to a {@link DiagnosticEvent}. Not all parameters may be applicable, depending on the
+ * type of DiagnosticEvent being instantiated.
+ */
+@Builder
+@Getter
+@Accessors(fluent = true)
+@EqualsAndHashCode
+@ToString
+public class DiagnosticEventInput {
```

Comment:
Can we have an event input for each of the event like`ExecutorStateEventInput` and `RejectedTaskEventInput extends ExecutorStateEventInput`

Reply:
Per Sahil's comments below we may want to rethink the inheritance design here and use interfaces instead

Reply:
For the sake of simplicity, just left simple constructors for now. Can refactor to an input wrapper if and when necessary